ui/pages/onboarding-flow/import-srp/import-srp.js

```diff
@@ -16,44 +14,58 @@ import {
   TYPOGRAPHY,
 } from '../../../helpers/constants/design-system';
 import { ONBOARDING_CREATE_PASSWORD_ROUTE } from '../../../helpers/constants/routes';
-import { clearClipboard } from '../../../helpers/utils/util';
 import { useI18nContext } from '../../../hooks/useI18nContext';
 import ZENDESK_URLS from '../../../helpers/constants/zendesk-url';
+import SrpInput from '../../../components/app/srp-input';
 
 export default function ImportSRP({ submitSecretRecoveryPhrase }) {
   const [secretRecoveryPhrase, setSecretRecoveryPhrase] = useState('');
-  const [revealSRP, setRevealSRP] = useState(false);
-  const [error, setError] = useState('');
   const history = useHistory();
   const t = useI18nContext();
-  const { isValidMnemonic } = ethers.utils;
 
-  const parseSeedPhrase = (seedPhrase) =>
-    (seedPhrase || '').trim().toLowerCase().match(/\w+/gu)?.join(' ') || '';
-
-  const handleSecretRecoveryPhraseChange = (recoveryPhrase) => {
-    setError('');
-    if (recoveryPhrase) {
-      const parsedSecretRecoveryPhrase = parseSeedPhrase(recoveryPhrase);
-      const wordCount = parsedSecretRecoveryPhrase.split(/\s/u).length;
-      if (wordCount % 3 !== 0 || wordCount > 24 || wordCount < 12) {
-        setError(t('seedPhraseReq'));
-      } else if (!isValidMnemonic(parsedSecretRecoveryPhrase)) {
-        setError(t('invalidSeedPhrase'));
-      }
+  /**
+   * Function checks if the button is disabled or enabled
+   *
+   * If secretRecoveryPhrase.length = 0 button will be disabled
+   * If we choose "I have 12-word phrase" in Dropdown component than,
+   * secretRecoveryPhrase.length = 11 and the button will be disabled
+   * If we choose "I have 15-word phrase" in Dropdown component than,
+   * secretRecoveryPhrase.length = 14 and the button will be disabled
+   * If we choose "I have 18-word phrase" in Dropdown component than,
+   * secretRecoveryPhrase.length = 17 and the button will be disabled
+   * If we choose "I have 21-word phrase" in Dropdown component than,
+   * secretRecoveryPhrase.length = 20 and the button will be disabled
+   * If we choose "I have 24-word phrase" in Dropdown component than,
+   * secretRecoveryPhrase.length = 23 and the button will be disabled
+   */
+  const checkButtonVisibility = () => {
```

Comment:
Is this function needed? The `SrpInput` component should already guarantee that the SRP is a valid length. The change handler is only called with either an empty string, or a guaranteed valid SRP.

Reply:
@Gudahtt This function is not needed, you are right. I checked again how to solve button visibility, when the button is disabled and enabled and I found better solution. [Here](https://github.com/MetaMask/metamask-extension/pull/14425/commits/b08521f00fbcc0ca018296084d50fbfe951269fa) are changes that I made for this button.